In [3]:
import pandas as pd

df = pd.read_csv('cleaned_mall_customers.csv')

print("First 5 rows:")
display(df.head())

print("\nBasic Statistics:")
display(df.describe())

First 5 rows:


,CustomerID,Genre,Age,Annual_Income_(k$),Spending_Score
0,1,Male,19,61.5,39
1,2,Male,21,15.0,81
2,3,Female,20,16.0,6
3,4,Female,23,16.0,77
4,5,Female,31,17.0,40



Basic Statistics:


,CustomerID,Age,Annual_Income_(k$),Spending_Score
count,200.000000,200.000000,200.000000,200.000000
mean,100.500000,38.850000,60.792500,50.200000
std,57.879185,13.969007,26.064439,25.823522
min,1.000000,18.000000,15.000000,1.000000
25%,50.750000,28.750000,42.000000,34.750000
50%,100.500000,36.000000,61.750000,50.000000
75%,150.250000,49.000000,78.000000,73.000000
max,200.000000,70.000000,137.000000,99.000000


In [ ]:
import matplotlib.pyplot as plt

fig, axes = plt.subplots(1, 2, figsize=(14, 5))

axes[0].hist(df['Age'], bins=10, color='skyblue', edgecolor='black')
axes[0].set_title('Age Distribution of Customers')
axes[0].set_xlabel('Age')
axes[0].set_ylabel('Count')
axes[0].grid(True)

axes[1].scatter(df['Annual_Income_(k$)'], df['Spending_Score'], color='purple', alpha=0.7, s=70)
axes[1].set_title('Annual Income vs. Spending Score')
axes[1].set_xlabel('Annual Income (k$)')
axes[1].set_ylabel('Spending Score (1-100)')
axes[1].grid(True)

plt.tight_layout()
plt.show()

Here are the key insights discovered from the Mall Customers dataset:

1.Age vs. Spending Score Relationship (Younger Shoppers Spend More)

*Observation: There is a noticeable negative correlation between age and spending score. Customers under the age of 40 exhibit significantly higher spending scores (ranging mostly between 60 and 100) compared to older demographics.

*Business Value: Marketing campaigns and promotional discounts should target younger demographics to drive higher sales conversion rates.

2.Customer Segmentation via Annual Income and Spending Score

*Observation: The scatter plot between Annual Income and Spending Score reveals distinct customer clusters (e.g., low-income high-spenders, high-income high-spenders, and moderate-income moderate-spenders).

*Business Value: These natural groupings provide a solid foundation for building K-Means clustering models to tailor personalized marketing strategies for each segment.

3.Gender Distribution Breakdown

*Observation: Female customers account for 56% (112) of the total customer base, whereas male customers make up 44% (88), showing a slightly higher proportion of female shoppers visiting the mall.

*Business Value: Helps in planning gender-specific retail inventories, product placements, and targeted promotional events.

4.High-Income Outliers Identification

*Observation: The annual income ranges up to $137k, identifying a specific group of high-earning outliers who have high purchasing power.

*Business Value: This niche group can be exclusively targeted with premium products, luxury goods, and exclusive VIP membership programs.

5.Balanced Central Tendency in Spending Behavior

*Observation: Both the mean and median of the Spending Score hover closely around 50.2, indicating a normally distributed middle ground where a large portion of customers maintain a balanced approach between saving and spending.

*Business Value: Essential baseline metric for general store-wide discounts and standard promotional campaigns that cater to the average consumer.